substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -118,6 +120,87 @@ impl IntoPortable for RuntimeApiMethodParamMetadataIR {
 	}
 }
 
+/// Metadata of the the runtime query dispatch.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct RuntimeQueryIR<T: Form = MetaForm> {
+	/// The type implementing the runtime query dispatch.
+	pub ty: T::Type,
+	/// The query interfaces metadata.
+	pub interfaces: Vec<QueryInterfaceIR<T>>,
+}
+
+/// Metadata of a runtime query interface.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct QueryInterfaceIR<T: Form = MetaForm> {
+	/// Name of the query interface.
+	pub name: T::String,
+	/// Queries belonging to the query interface.
+	pub queries: Vec<QueryMetadataIR<T>>,
+	/// Query interface documentation.
+	pub docs: Vec<T::String>,
+}
+
+impl IntoPortable for QueryInterfaceIR {
+	type Output = QueryInterfaceIR<PortableForm>;
+
+	fn into_portable(self, registry: &mut Registry) -> Self::Output {
+		QueryInterfaceIR {
+			name: self.name.into_portable(registry),
+			queries: registry.map_into_portable(self.queries),
+			docs: registry.map_into_portable(self.docs),
+		}
+	}
+}
+
+/// Metadata of a runtime method.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct QueryMetadataIR<T: Form = MetaForm> {
```

Comment:
dq: but don't we want to have space for version/deprecation info on the query to allow iteration on already defined apis without explicitly removing them or redefining the type signature? 
Or the intention is just o keep them all and fully rely on `QueryId` for the resolution?

Reply:
The idea was not to have versioning, a new version would just have a different `QueryId`. The old version would have to be maintained possibly indefinitely. Perhaps a deprecation flag would be useful, needs some thinking about.